### PR TITLE
MM-10140: API Implementation for Schemes related Endpoints

### PR DESCRIPTION
**Pull request by grundleborg**
#### Summary
This implements the general purpose APIs for Schemes, as described in the API Reference PR [here](https://github.com/mattermost/mattermost-api-reference/pull/339).

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-10140

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation: https://github.com/mattermost/mattermost-api-reference/pull/339
- [x] All new/modified APIs include changes to the drivers
- [x] Includes text changes and localization file updates

### CODE REVIEW

File: api4/api.go [hunk 1]
```diff
@@ -99,7 +99,8 @@ type Routes struct {
 
 	Reactions *mux.Router // 'api/v4/reactions'
 
-	Roles *mux.Router // 'api/v4/roles'
+	Roles   *mux.Router // 'api/v4/roles'
+	Schemes *mux.Router // 'api/v4/roles'
```
Comment (jespino) [inline api4/api.go]: the comment is wrong

File: api4/scheme.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+	}
+}
+
+func getScheme(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireSchemeId()
+	if c.Err != nil {
+		return
+	}
+
+	if scheme, err := c.App.GetScheme(c.Params.SchemeId); err != nil {
```
Comment (jespino) [inline api4/scheme.go]: Looks like every endpoint requires PERMISSION_MANAGE_SYSTEM except the getScheme, which only need the schemeId. I'm not 100% sure this is what is expected.

File: api4/scheme.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+		return
+	}
+
+	if !c.App.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+		return
+	}
+
+	scope := c.Params.Scope
+	if len(scope) > 0 && !(scope == model.SCHEME_SCOPE_TEAM || scope == model.SCHEME_SCOPE_CHANNEL) {
```
Comment (jespino) [inline api4/scheme.go]: This if is harder to understand than something like `scope != "" && scope != model.SCHEME_SCOPE_TEAM && scope != model.SCHEME_SCOPE_CHANNEL`

File: api4/scheme_test.go [hunk 4]
```diff
@@ -0,0 +1,661 @@
...
+	scheme2 := &model.Scheme{
+		Name:        model.NewId(),
+		Description: model.NewId(),
+		Scope:       model.SCHEME_SCOPE_CHANNEL,
+	}
+	scheme2, rs2 := th.SystemAdminClient.CreateScheme(scheme2)
+	CheckNoError(t, rs2)
+
+	_, ri5 := th.SystemAdminClient.GetTeamsForScheme(scheme2.Id, 0, 100)
+	CheckBadRequestStatus(t, ri5)
```
Comment (jespino) [inline api4/scheme_test.go]: Maybe make sense to use here in in other tests similar to this one Table Based Testing? Only a suggestion, of course.

File: api4/scheme.go (hunk 2 above)
Comment (grundleborg) [reply api4/scheme.go]: I think this is intended, at least until there's a specific reason for any of the others to be used outside the admin console.

File: api4/scheme_test.go (hunk 4 above)
Comment (grundleborg) [reply api4/scheme_test.go]: Yeah, I guess I could probably have done that. Not going to bother changing it now though - next time...

File: api4/scheme.go (hunk 2 above)
Comment (jespino) [reply api4/scheme.go]: My question is, which users needs to access to the scheme information?

Comment (grundleborg) [reply api4/scheme.go]: Oh I see, fair point. I'll add a permission check too here.

File: api4/scheme.go [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+		c.Err = err
+		return
+	}
+
+	if scheme.Scope != model.SCHEME_SCOPE_TEAM {
+		c.Err = model.NewAppError("Api4.GetTeamsForScheme", "api.scheme.get_teams_for_scheme.scope.error", nil, "", http.StatusBadRequest)
+		return
+	}
+
+	if teams, err := c.App.GetTeamsForSchemePage(scheme, c.Params.Page, c.Params.PerPage); err != nil {
```
Comment (mkraft) [inline api4/scheme.go]: 0/5 but I had to do a double-take to see whether `GetTeamsForSchemePage` took a "SchemePage" as a parameter vs if this were called `GetTeamsPageForScheme`.

File: api4/scheme.go (hunk 2 above)
Comment (jespino) [reply api4/scheme.go]: Remember to update the API documentation too

**Pull request merged**